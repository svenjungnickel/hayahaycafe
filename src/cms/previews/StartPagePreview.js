import React from 'react';
import PropTypes from 'prop-types';
import StartPageTemplate from '../../templates/StartPageTemplate';

const StartPagePreview = ({ entry, widgetFor }) => (
    <StartPageTemplate
        title={entry.toJS().data.title}
        subtitle={entry.toJS().data.subtitle}
        headerImage={entry.toJS().data.headerImage}
        content={widgetFor('body')}
        location={widgetFor('location')}
        openingHours={widgetFor('openingHours')}
    />
);

StartPagePreview.propTypes = {
    entry: PropTypes.shape({
        getIn: PropTypes.func,
    }),
    widgetFor: PropTypes.func,
};

export default StartPagePreview;
