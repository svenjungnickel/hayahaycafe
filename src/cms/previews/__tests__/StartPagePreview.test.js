import React from 'react';
import renderer from 'react-test-renderer';
import StartPagePreview from '../StartPagePreview';

jest.mock('../../../components/Image');
jest.mock('../../../components/Gallery');
jest.mock('../../../components/Header');
jest.mock('../../../components/Location/Location');
jest.mock('../../../components/Separator');
jest.mock('../../../components/OpeningHours');
jest.mock('../../../components/What3WordsAddress');

describe('StartPagePreview', () => {
    it('renders without entry throws prop type error', () => {
        const renderStartPagePreview = () => {
            const widgetFor = () => {};

            const component = <StartPagePreview widgetFor={widgetFor} />;
            renderer.create(component);
        };

        expect(renderStartPagePreview).toThrowError('Warning: Failed prop type');
    });

    it('renders without widgetFor throws prop type error', () => {
        const renderDefaultPagePreview = () => {
            const entry = {
                toJS: () => {},
            };

            const component = <StartPagePreview entry={entry} />;
            renderer.create(component);
        };

        expect(renderDefaultPagePreview).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid entry throws prop type error', () => {
        const renderDefaultPagePreview = () => {
            const entry = {
                toJS: 123,
            };
            const widgetFor = () => {};

            const component = <StartPagePreview entry={entry} widgetFor={widgetFor} />;
            renderer.create(component);
        };

        expect(renderDefaultPagePreview).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid widgetFor throws prop type error', () => {
        const renderDefaultPagePreview = () => {
            const entry = {
                toJS: () => {},
            };
            const widgetFor = {};

            const component = <StartPagePreview entry={entry} widgetFor={widgetFor} />;
            renderer.create(component);
        };

        expect(renderDefaultPagePreview).toThrowError('Warning: Failed prop type');
    });

    it('renders correctly', () => {
        const entry = {
            toJS: () => {
                return {
                    data: {
                        title: 'title',
                        subtitle: 'subtitle',
                        headerImage: {
                            publicURL: 'publicURL',
                        },
                        gallery: [
                            {
                                src: {
                                    publicURL: 'publicURL',
                                },
                                alt: 'gallery image 01',
                                title: 'Gallery Image 01',
                            },
                            {
                                src: {
                                    publicURL: 'publicURL',
                                },
                                alt: 'gallery image 02',
                                title: 'Gallery Image 02',
                            },
                        ],
                    },
                };
            },
        };
        const widgetFor = () => 'content';

        const component = <StartPagePreview entry={entry} widgetFor={widgetFor} />;
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
