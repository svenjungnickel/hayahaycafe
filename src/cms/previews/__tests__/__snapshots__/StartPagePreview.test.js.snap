// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`StartPagePreview renders correctly 1`] = `
[
  <header
    className="startHeader"
  >
    <img
      alt="alt"
      src="src"
    />
    <div
      className="startHeader__content"
    >
      <div
        className="container"
      >
        <div
          className="startHeader__contentInner"
        >
          <h1
            className="display-1"
          >
            title
          </h1>
          <p
            className="lead"
          >
            subtitle
          </p>
        </div>
      </div>
    </div>
  </header>,
  <section
    className="section"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <img
            alt="alt"
            src="src"
          />
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <div>
            content
          </div>
          <hr />
        </div>
        <div
          className="col-sm-4 col-12"
          data-cy="startPageLocation"
        >
          <p>
            Location
          </p>
        </div>
        <div
          className="col-sm-4 col-12"
          data-cy="startPageWhat3Words"
        >
          <hr />
          <a
            href="what3WordsAddress"
          >
            <img
              alt="what3WordsIcon"
              src="src"
            />
          </a>
        </div>
        <div
          className="col-sm-4 col-12"
          data-cy="startPageOpeningHours"
        >
          <hr />
          <p>
            Opening Hours
          </p>
        </div>
      </div>
    </div>
  </section>,
]
`;
